Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDll.cs

```diff
@@ -75,6 +75,23 @@ internal class DbgEngDll : IDebuggerEngine
         private static IDebugSystemObjects4 systemObjects;
 
         /// <summary>
+        /// Dictionary of all debugee flow controlers. Key is process id that is being debugged.
+        /// </summary>
+        private static DictionaryCache<uint, DebuggeeFlowController> debugeeFlowControlers;
+
+        /// <summary>
+        /// Static constructor.
+        /// </summary>
+        static DbgEngDll()
+        {
+            // Populate flow controlers lazely.
+            //
+            debugeeFlowControlers =
+                new DictionaryCache<uint, DebuggeeFlowController>(
+                    (processId) => new DebuggeeFlowController(originalClient));
```

Comment:
Maybe it is easier to follow if you actually use DbgEng class to access client. It provides you all the necessary interfaces as static properties...